### PR TITLE
Block clean install of OCP 3.8 and upgrade to 3.9 only

### PR DESCRIPTION
**Pull request by vrutkovs**
This PR ensures clean 3.8 cannot be installed and 3.7 can be upgraded
to 3.9 only.

TODO:
- [x] Decide on Origin 3.8 vs. OCP 3.8.
  ~~Sounds like we certainly don't want OCP 3.8 upgrade and clean install, does it also apply to Origin?~~
   Neither OCP not Origin 3.8 are supported, playbooks were removed
- [x] Block clean OCP 3.8 install of any component
- [x] Check during control plane upgrade that both 3.8 and 3.9 packages
      are available on all masters if RPM based install. 
      Abort if both aren't available.

      This is already implemented, just a small fix to show which 
      package version is missing

- [x] Update upgrades/README.md about supported versions
- [x] Block node upgrade from 3.7 to 3.8.

      Playbooks to upgrade node from 3.7 to 3.8 were removed,
      nodes won't be updated to 3.8, but straight to 3.9 for OCP.

Note, that 3.8 upgrade playbooks were not removed to ensure 3.8 -> 3.9 
upgrade can still be performed in case any failure occurs.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1541555

### CODE REVIEW

**Comment by vrutkovs** [comment]
Looks like some weird flake

/test gcp

File: playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml [hunk 1]
```diff
@@ -266,8 +266,9 @@
   - include_tasks: docker/tasks/upgrade.yml
     when: l_docker_upgrade is defined and l_docker_upgrade | bool and not openshift_is_atomic | bool
 
+
 - name: Drain and upgrade master nodes
-  hosts: oo_masters_to_config:&oo_nodes_to_upgrade
+  hosts: "{{ l_double_upgrade_cp | default(False) | ternary('all:!all', 'oo_masters_to_config:&oo_nodes_to_upgrade') }}"
```
Comment (michaelgugino) [inline playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml]: Let's add a comment why were doing this here.

File: playbooks/common/openshift-cluster/upgrades/v3_8/upgrade.yml [hunk 2]
```diff
@@ -54,5 +54,6 @@
       state: started
 
 - import_playbook: ../upgrade_nodes.yml
```
Comment (michaelgugino) [inline playbooks/common/openshift-cluster/upgrades/v3_8/upgrade.yml]: Why are we using this boolean here?

Comment (vrutkovs) [reply playbooks/common/openshift-cluster/upgrades/v3_8/upgrade.yml]: Added a new TODO, as I assumed only OCP 3.8 should be restricted, but not Origin

File: playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml (hunk 1 above)
Comment (vrutkovs) [reply playbooks/common/openshift-cluster/upgrades/upgrade_control_plane.yml]: Done

File: playbooks/common/openshift-cluster/upgrades/v3_8/upgrade.yml (hunk 2 above)
Comment (vrutkovs) [reply playbooks/common/openshift-cluster/upgrades/v3_8/upgrade.yml]: Scratch that, neither Origin nor OCP 3.8 are supported

**Comment by vrutkovs** [comment]
Flakes

/retest

**Pull request merged**